hornetq-server/src/main/java/org/hornetq/core/server/impl/QueueImpl.java

```diff
@@ -2181,7 +2180,7 @@ private void checkDepage()
     */
    private boolean needsDepage()
    {
-      return queueMemorySize.get() < pageSubscription.getPagingStore().getMaxSize();
+      return queueMemorySize.get() < pageSubscription.getPagingStore().getMaxSize() || pageSubscription.getPagingStore().getMaxSize() == -1;
```

Comment:
Artemis?

Reply:
I don't want to merge fixes in HornetQ if they don't have an equivalent fix in Artemis... it would get out of control easily.

Reply:
@clebertsuconic I investigated the Artemis case and found that it doesn't have this kind of issue anymore. It happens that Artemis fixes this issue by a commit fixing Artemis-581. The part that fixes this particular issue is in PagingStoreImpl.java:

@Override
    public long getMaxSize() {
-      return maxSize;
+      if (maxSize < 0) {
+         // if maxSize < 0, we will return 2 pages for depage purposes
+         return pageSize * 2;
+      }
+      else {
+         return maxSize;
+      }
    }
I'm not sure what's the purpose for fixing Artemis-581, but this avoids this HornetQ issue.
So basically Artemis don't need the fix. Only I think the test is valid and can be ported into Artemis.

Howard

Reply:
@clebertsuconic Can you merge this PR pls?